pep-0432.txt

```diff
@@ -6,13 +6,48 @@ Author: Nick Coghlan <ncoghlan@gmail.com>,
         Victor Stinner <vstinner@python.org>,
         Eric Snow <ericsnowcurrently@gmail.com>
 Discussions-To: capi-sig@python.org
-Status: Draft
+Status: Withdrawn
 Type: Standards Track
 Content-Type: text/x-rst
 Requires: 587
 Created: 28-Dec-2012
-Python-Version: 3.9
-Post-History: 28-Dec-2012, 2-Jan-2013, 30-Mar-2019
+Post-History: 28-Dec-2012, 2-Jan-2013, 30-Mar-2019, 28-Jun-2020
+
+
+PEP Withdrawal
+==============
+
+From late 2012 to mid 2020, this PEP provided general background and specific
+concrete proposals for making the CPython startup sequence easier to maintain
+and the CPython runtime easier to embed as part of a larger application.
+
+For most of that time, the changes were maintained either in a separate feature
+branch, or else as underscore-prefixed private APIs in the main CPython repo.
+
+In 2019, PEP 587 migrated a subset of those API changes to the public CPython
+API for Python 3.8+.
+
+In June 2020, in response to a query from the Steering Council, the PEP authors
+decided that it made sense to withdraw the original PEP, as enough has changed
+since PEP 432 was first written that we think any further changes to the
+startup sequence and embedding API would be best formulated as a new PEP (or
+PEPs) that take into account not only the not-yet-implemented ideas from PEP 432
+that weren't considered sufficiently well validated to make their way into
+PEP 587, but also any feedback on the public PEP 587 API, and any other lessons
+that have been learned while adjusting the CPython implementation to be more
+embedding and subinterpreter friendly.
+
+In particular, PEPs proposing the following changes, and any further
+infrastructure changes needed to enable them, would likely still be worth
+exploring:
+
+* shipping an alternate Python executable that ignores all user level
+  settings and runs in isolated mode by default, and would hence be more
+  suitable for execution of system level Python applications than the default
+  interpreter
+* enhancing the zipapp module to support the creation of single-file executables
+  from pure Python scripts (and potentially even Python extension modules, given
+  the introduction of multi-phase extension module initialisation)
```

Comment:
Hum, one of the PEP 432 motivation was to rewrite the getpath.c code. IMHO it's still a valid point :-)